src/main/java/org/testng/TestNG.java

```diff
@@ -292,12 +293,19 @@ private void parseSuite(String suitePath) {
       for (XmlSuite s : allSuites) {
         s.setParallel(this.m_parallelMode);
         s.setThreadCount(this.m_threadCount);
-        // If test names were specified, only run these test names
-        if (m_testNames != null) {
-          m_suites.add(XmlSuiteUtils.cloneIfContainsTestsWithNamesMatchingAny(s, m_testNames));
-        } else {
+        if (m_testNames == null) {
           m_suites.add(s);
+          continue;
         }
+        // If test names were specified, only run these test names
+        XmlSuiteUtils xmlSuiteUitls = new XmlSuiteUtils();
+        xmlSuiteUitls.cloneIfContainsTestsWithNamesMatchingAny(s, m_testNames);
+        List<String> missMatchedTestname = xmlSuiteUitls.getMissMatchedTestNames(m_testNames);
+        if (CollectionUtils.hasElements(missMatchedTestname)) {
```

Comment:
No need of `CollectionUtils`, `missMatchedTestname` must not be null.

Reply:
yes, you are right.